xarray/core/variable.py

```diff
@@ -1069,6 +1071,32 @@ def chunk(self, chunks={}, name=None, lock=False):
 
         return self._replace(data=data)
 
+    def to_numpy(self) -> np.ndarray:
+        """Coerces wrapped data to numpy and returns a numpy.ndarray"""
+        # TODO an entrypoint so array libraries can choose coercion method?
+        data = self.data
+        try:
+            data = data.to_numpy()
+        except AttributeError:
+            if isinstance(data, dask_array_type):
+                data = data.compute()
+            if isinstance(data, cupy_array_type):
+                data = data.get()
+            # pint has to be imported dynamically as pint imports xarray
+            pint_array_type = DuckArrayModule("pint").type
+            if isinstance(data, pint_array_type):
+                data = data.magnitude
+            if isinstance(data, sparse_array_type):
+                data = data.todense()
+            if type(data) != np.ndarray:  # noqa : Don't allow subclasses
+                data = np.asarray(data)
```

Comment:
No real need for the conditional check here, since `asarray()` already does the right thing on numpy arrays:
```suggestion
            data = np.asarray(data)
```